src/code/audio_load.c

```diff
@@ -885,145 +921,141 @@ s32 Audio_DMA(OSIoMesg* mesg, u32 priority, s32 direction, u32 devAddr, void* ra
+                    return ret;
                 }
                 break;
             case 3:
             case 4:
-                sp4C = Audio_AllocBankOrSeq(tableType, size, 2, sp34);
-                if (sp4C == NULL) {
-                    return sp4C;
+                ret = AudioHeap_AllocCached(tableType, size, CACHE_EITHER, realId);
+                if (ret == NULL) {
+                    return ret;
                 }
                 break;
         }
 
-        if (sp48 == 1) {
-            func_800E3A44((s16)sp50->header.unk_02, devAddr, sp4C, size, sp48, arg2, retQueue,
-                          (arg3 << 0x18) | (tableType << 0x10) | (arg1 << 8) | sp3C);
+        if (medium == MEDIUM_UNK) {
+            AudioLoad_StartAsyncLoadUnkMedium((s16)sp50->unkMediumThing, devAddr, ret, size, medium, nChunks, retQueue,
+                                              MK_ASYNC_MSG(retData, tableType, id, status));
         } else {
-            Audio_InitAsyncReq(devAddr, sp4C, size, sp48, arg2, retQueue, MK_ASYNC_MSG(arg3, tableType, sp34, sp3C));
+            AudioLoad_StartAsyncLoad(devAddr, ret, size, medium, nChunks, retQueue,
+                                     MK_ASYNC_MSG(retData, tableType, realId, status));
         }
-        sp3C = 1;
+        status = 1;
     }
 
     switch (tableType) {
-        case 0:
-            Audio_SetSeqLoadStatus(sp34, sp3C);
+        case SEQUENCE_TABLE:
+            AudioLoad_SetSeqLoadStatus(realId, status);
             break;
-        case 1:
-            Audio_SetBankLoadStatus(sp34, sp3C);
+        case FONT_TABLE:
+            AudioLoad_SetFontLoadStatus(realId, status);
             break;
-        case 2:
-            func_800E1A78(sp34, sp3C);
+        case SAMPLE_TABLE:
+            AudioLoad_SetSampleFontLoadStatusAndApplyCaches(realId, status);
             break;
         default:
             break;
     }
 
-    return sp4C;
+    return ret;
 }
 
-void Audio_ProcessLoads(s32 resetStatus) {
-    Audio_ProcessSyncLoads(resetStatus);
-    func_800E4590(resetStatus);
-    Audio_ProcessAsyncLoads(resetStatus);
+void AudioLoad_ProcessLoads(s32 resetStatus) {
+    AudioLoad_ProcessSlowLoads(resetStatus);
+    AudioLoad_ProcessSamplePreloads(resetStatus);
+    AudioLoad_ProcessAsyncLoads(resetStatus);
 }
 
-// SetDmaCallback
-void func_800E301C(void* callback) {
+void AudioLoad_SetDmaHandler(DmaHandler callback) {
     sDmaHandler = callback;
 }
 
-u32 D_801304D4 = 0;
-void func_800E3028(u32 arg0) {
-    D_801304D4 = arg0;
+void AudioLoad_SetUnusedHandler(void* callback) {
+    sUnusedHandler = callback;
 }
 
-// InitCtlTable
-void func_800E3034(s32 arg0) {
-    CtlEntry* ctlEnt = &gAudioContext.ctlEntries[arg0];
-    AudioBankTableEntry* tableEnt = &gAudioContext.audioBankTable->entries[arg0];
+void AudioLoad_InitCtlEntry(s32 fontId) {
+    CtlEntry* entry = &gAudioContext.ctlEntries[fontId];
+    AudioTableEntry* tableEntry = &gAudioContext.soundFontTable->entries[fontId];
```

Comment:
```suggestion
void AudioLoad_InitFontEntry(s32 fontId) {
    CtlEntry* entry = &gAudioContext.ctlEntries[fontId];
    AudioTableEntry* tableEntry = &gAudioContext.soundFontTable->entries[fontId];
```
I guess any function with `Ctl` would need to change too